dataset/src/main/scala/frameless/ops/SortableColumnTypes.scala

```diff
@@ -0,0 +1,24 @@
+package frameless.ops
+
+import frameless.{CatalystOrdered, SortedTypedColumn}
+import shapeless.{::, HList, HNil}
+
+import scala.annotation.implicitNotFound
+
+@implicitNotFound(
+  "Either one of the selected columns is not sortable (${U}), " +
+    "or no ordering (ascending/descending) has been selected. " +
+    "Select an ordering on any column (t) using the t.asc or t.desc methods.")
+trait SortableColumnTypes[T, U <: HList]
```

Comment:
I think there might be something in shapeless doing just that.

Reply:
Got it, it's called [Comapped](https://github.com/milessabin/shapeless/blob/66d1b932e6e817999090ec5844b2b76bb8559d5f/core/src/main/scala/shapeless/ops/hlists.scala#L85)

Reply:
Thank you for all the comments @OlivierBlanvillain. This one does two things, enforces that they are CatalystOrdered and that they are SortedTypedColumn. It also allows us to give the ability to write a custom error compilation error. FInally, our ColumnTypes already is a small rework of Comapped and there @kanterov mentioned a performance reason why we prefer our own version rather than relying on Comapped.